### DISCUSSION TITLE
Plugin: couldn't get dashboard uid and panel id

### DISCUSSION THREAD

**Discussion by loserbird** [grafana/grafana#45315] [Data Sources]
What happened:
i would like to make a datasource plugin and i want to know which dashboard and which panel a slow query request comes from. However,there seems to be no way to get the dashboard uid and panel id from datasource plugin api. 

Proposed solution:
- It might be a good idea to put the dashboard UID and panel id into the http request headers and transfer them to headers of backend.QueryDataRequest in query.handleQueryData Method. 
![image](https://user-images.githubusercontent.com/26373043/153607531-7a24427b-0dde-4144-be79-1b9b39d05eef.png)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
